### PR TITLE
Add support for dynamic areas

### PR DESCRIPTION
**Pull request by mraspaud**

### CODE REVIEW

File: pyresample/geometry.py [hunk 1]
```diff
@@ -455,6 +456,118 @@ def __init__(self, lons, lats, nprocs=1):
...
+            proj_dict = self._compute_omerc_parameters(ellipsoid)
+        else:
+            raise NotImplementedError('Only omerc supported for now.')
+
+        area = DynamicAreaDefinition(area_id, description, proj_dict)
+        lons, lats = self.get_edge_lonlats()
+        return area.freeze(lons, lats, size=(x_size, y_size))
+
+
+class DynamicAreaDefinition():
```
Comment (davidh-ssec) [inline pyresample/geometry.py]: No base class?

Comment (mraspaud) [reply pyresample/geometry.py]: No I don't think it's needed. Instances of this class are never meant to be full fledged area that have resample capability.

File: pyresample/geometry.py [hunk 2]
```diff
@@ -449,12 +457,175 @@ class SwathDefinition(CoordinateDefinition):
...
+        area = DynamicAreaDefinition(area_id, description, proj_dict)
+        lons, lats = self.get_edge_lonlats()
+        return area.freeze((lons, lats), size=(x_size, y_size))
+
+
+class DynamicAreaDefinition():
+    """An AreaDefintion containing just a subset of the needed parameters.
+
+    The purpose of this class is to be able to adapt the area extent and size of
+    the area to a give set of longitudes and latitudes, such that e.g. polar
```
Comment (adybbroe) [inline pyresample/geometry.py]: "the area to a give set" -> change order of "a" and "give" :-)

File: pyresample/test/test_geometry.py [hunk 3]
```diff
@@ -701,6 +597,179 @@ def test_get_xy_from_lonlat(self):
...
+        area = geometry.SwathDefinition(lons, lats)
+        proj_dict = {'no_rot': True, 'lonc': 5.340645620216994,
+                     'ellps': 'WGS84', 'proj': 'omerc',
+                     'alpha': 19.022450179020247, 'lat_0': 60.7420043944989}
+        self.assertDictEqual(area._compute_omerc_parameters('WGS84'),
+                             proj_dict)
+
+    def test_get_edge_lonlats(self):
+        """Test the `get_edge_lonlats` functionality."""
+        lats = np.array([[85.23900604248047, 62.256004333496094, 35.58000183105469],
```
Comment (adybbroe) [inline pyresample/test/test_geometry.py]: Do you test an area containing the North/South pole?

Comment (mraspaud) [reply pyresample/test/test_geometry.py]: Not yet :)

File: pyresample/geometry.py (hunk 2 above)
Comment (mraspaud) [reply pyresample/geometry.py]: ack

**Pull request merged**